Auto-roll billing start date docs (#27685)
hashicorp/vault: website/content/docs/concepts/billing-start-date.mdx

---
layout: docs

> page_title: Billing Start Date

Reviewer: Same goes for other instances, such as L8 below and the changes in website/data/docs-nav-data.json.

description: >-
   A description of Vault cluster's billing start date used throughout Vault.
---

# Billing Start Date

> This is the billing start date associated with the reporting cluster. This will be the license start date if not configured.

Reviewer: ```suggestion
Billing start date reflects the current Vault license `StartTime` field.

Vault uses the billing start date as a default point of reference for license utilization queries, including:
1. Direct queries via the activitylog API (TODO: link to API docs)
2. activity export CLI
3. Automated License Utilization reports (link, if relevant)
4. Manual License Utilization Reports (link)
```

Author: The billing start date can also be configured via the config file. We do not show this field in documentation. I am not sure if we need to cover the config case here too.

Reviewer 2: Optimally, we would correct the gap in the config docs and then document the full options in a related how-to guide

Author: But I think we don't want to document about the billing start time field in config. I don't know when that decision was made probably when the initial census changes went out.

> This is the billing start date associated with the reporting cluster. This will be the license start date if not configured.

Reviewer 2: It would be useful if we actually defined what the billing start date is as the opening sentence. Saying that billing start date is the "billing start date associated with the reporting cluster" doesn't really tell me anything useful as a reader.

> As of 1.15.13, 1.16.7, 1.17.3, 1.18 and later, the billing start date will automatically rollover to the most recent year. 

Reviewer: We probably want to confirm release versions here, and maybe:

```suggestion
As of 1.15.13, 1.16.7, 1.17.3, 1.18+, the billing start date is automatically adjusted to the most recent billing year. For more information on automatic yearly billing rollover, refer to: [TODO: link to docs].
```

Author: What doc link are you referring to here? This is the main concepts doc

> As of 1.15.13, 1.16.7, 1.17.3, 1.18+, Vault clusters will automatically adjust the billing start date to provide a year's worth of utilization data.

Reviewer 2: ```suggestion
As of 1.15.13, 1.16.7, 1.17.3, 1.18+, Vault clusters will automatically adjust the billing start date to provide a year's worth of utilization data.
```
```suggestion
To provide a one-year window of utilization data, Vault clusters automatically adjust the billing start date [to | by] {DESCRIPTION_OF_ADJUSTMENT}.
```
I'm still not clear exactly *how* we adjust the billing start date (hence the "variable" in the suggestion.

As a general note, except when we're calling out bugs/important changes in things like release notes or changelogs, docs are versions to specific Vault releases. The docs in `main` should always focus on the the current Vault version rather than calling out multiple versions. 

Since we've limited backport behavior to the most current version of Vault, that means we need to directly update the docs for any other versions with separate PRs (either by directly creating a new PR or manually backporting changes in another PR).
